pkgs/by-name/sd/SDL_compat/package.nix

```
@@ -52,8 +58,33 @@ stdenv.mkDerivation (finalAttrs: {
     ]
     ++ lib.optionals openglSupport [ libGLU ];
 
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace-fail 'set(CMAKE_SKIP_RPATH TRUE)' 'set(CMAKE_SKIP_RPATH FALSE)'
+  '';
+
+  dontPatchELF = true; # don't strip rpath
+
+  cmakeFlags =
+    let
+      rpath = lib.makeLibraryPath [ sdl2-compat ];
+    in
+    [
+      (lib.cmakeFeature "CMAKE_INSTALL_RPATH" rpath)
+      (lib.cmakeFeature "CMAKE_BUILD_RPATH" rpath)
+      (lib.cmakeBool "SDL12TESTS" finalAttrs.finalPackage.doCheck)
+    ];
+
   enableParallelBuilding = true;
 
+  # Darwin fails with "Critical error: required built-in appearance SystemAppearance not found"
+  doCheck = !stdenv.hostPlatform.isDarwin;
+  checkPhase = ''
+    runHook preCheck
+    ./testver
+    runHook postCheck
+  '';
```

```suggestion
  nativeInstallCheckInputs = [
    versionCheckHook
  ];
  doInstallCheck = true;
```

The version check hook is probably the more-correct way of doing this check

---

Okay fair! Just feels a bit weird: `sdl-config` has a `--version` flag you can give it, so `versionCheckHook` do the same thing at a glance. Might need a comment explaining you want the example program explicitly, not simply a version check.